### ISSUE TITLE
Error using XChaCha20-Poly1305

### ISSUE THREAD

**Issue by ScottG76** [dint-dev/cryptography#66]
The following error occurs using XChaCha20-Poly1305; however, ChaCha20-Poly1305 works fine.

```
  Invalid argument (nonce): Nonce must have 12 bytes: Uint8Array
    package:cryptography/src/dart/chacha20.dart 112:7                 DartChacha20.encrypt
    package:cryptography/src/dart/chacha20_poly1305_aead.dart 150:39  DartChacha20Poly1305AeadMacAlgorithm._poly1305SecretKeyFromChacha20
    package:cryptography/src/dart/chacha20_poly1305_aead.dart 59:40   DartChacha20Poly1305AeadMacAlgorithm.calculateMac
    package:cryptography/src/dart/xchacha20.dart 146:36               DartXchacha20.encrypt
```

Example:
```
      final crypto = Cryptography.instance.xchacha20Poly1305Aead();
      final secret = await Sha256().hash(utf8.encode("password"));

      await crypto.encrypt(
        utf8.encode("test"),
        secretKey: await crypto.newSecretKeyFromBytes(secret.bytes),
      );
```

**Comment by ScottG76** [comment]
Using version 'cryptography: ^2.0.2'

**Comment by terrier989** [comment]
Thanks for the issue report! This was fixed in the version 2.0.5.
